src/frontend/flight/handler.rs

```diff
@@ -163,32 +163,38 @@ impl SeafowlFlightHandler {
             });
         }
 
-        if cmd.format != TableFormat::Delta as i32 {
+        let (sync_target, url) = if cmd.format == TableFormat::Delta as i32 {
+            let log_store = match cmd.store {
+                None => self
+                    .context
+                    .get_internal_object_store()?
+                    .get_log_store(&cmd.path),
+                Some(store_loc) => {
+                    self.context
+                        .metastore
+                        .object_stores
+                        .get_log_store_for_table(
+                            Url::parse(&store_loc.location).map_err(|e| {
+                                DataFusionError::Execution(format!(
+                                    "Couldn't parse sync location: {e}"
+                                ))
+                            })?,
+                            store_loc.options,
+                            cmd.path,
+                        )
+                        .await?
+                }
+            };
+            let url = log_store.root_uri();
+            (LakehouseSyncTarget::Delta(log_store), url)
+        } else if cmd.format == TableFormat::Iceberg as i32 {
             return Err(SyncError::NotImplemented);
-        }
-
-        let log_store = match cmd.store {
-            None => self
-                .context
-                .get_internal_object_store()?
-                .get_log_store(&cmd.path),
-            Some(store_loc) => {
-                self.context
-                    .metastore
-                    .object_stores
-                    .get_log_store_for_table(
-                        Url::parse(&store_loc.location).map_err(|e| {
-                            DataFusionError::Execution(format!(
-                                "Couldn't parse sync location: {e}"
-                            ))
-                        })?,
-                        store_loc.options,
-                        cmd.path,
-                    )
-                    .await?
-            }
+        } else {
+            return Err(SyncError::InvalidMessage {
+                reason: "Unknown table format".to_string(),
```

Comment:
I think you might be able to match exactly on `cmd.format()` and avoid this part.